### ISSUE TITLE
[service] The service is building packages that are Outdated from their recipes

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan-center-index#200]
Using revisions, that shouldn't be possible.

To check:

```bash
conan search zlib/1.2.11@ --table=file.html -r=conan-center
```

**Comment by lasote** [comment]
Not happening in the internal artifactory repo. To investigate after the migration of the regenerated packages is completed.

**Comment by lasote** [comment]
Looks like it might be an issue with the `search` used without revisions. 

- If you enable revisions, there are no outdated packages.
- It only happens with Bintray, Artifactory works ok.
- It doesn't affect the install, which resolves to the correct latest revision.

**Comment by czoido** [comment]
I have been doing some tests and it looks like each time you search for the packages with revisions disabled in the client, the remote it's returning random values for the `outdated` field.

For the exact same search consecutively:

`conan search libcurl/7.67.0@ --table=file.html -r=conan-center`

![image](https://user-images.githubusercontent.com/5045666/71099156-1d244400-21b3-11ea-90c6-6ccfde41ef32.png)
![image](https://user-images.githubusercontent.com/5045666/71099171-244b5200-21b3-11ea-9a36-cfde0ce6de69.png)
![image](https://user-images.githubusercontent.com/5045666/71099190-2c0af680-21b3-11ea-9866-5adba13a4e25.png)
It's not a problem of the html generation, the results in the terminal are random as well.

**Comment by czoido** [comment]
Fortunately for the `install` command it always points to the latest revision:
Doing a test with a dummy package with lots of revisions:

```
➜  test_package conan search dummy_lib_test_revs/0.1@ -r=czoido-bintray --revisions
Revisions for 'dummy_lib_test_revs/0.1' at remote 'czoido-bintray':
c451b46ca01e0f132ef819008ea2dce4 (2019-12-19 10:47:27 UTC)
d10ce6dd81b5f845e103efd61ac96666 (2019-12-19 10:47:07 UTC)
d619c56f58dc764d59fccc28b673ccd5 (2019-12-19 10:46:46 UTC)
cc08e86826628b39ff99fc33754817df (2019-12-19 10:46:26 UTC)
e95cb3f51ddfa49b7572698a673b1479 (2019-12-19 10:46:04 UTC)
4107524b4579de56f2ba70e902a773a9 (2019-12-19 10:44:57 UTC)
75e7bacd94bf45fb244cf1d527d61883 (2019-12-19 10:44:36 UTC)
854ddcae6fa4091b184bf93f55203ef6 (2019-12-19 10:44:14 UTC)
4235169107ae00eb15a3850bf2e985c9 (2019-12-19 10:43:53 UTC)
611bcd42fba20cc222989ba04b9181b9 (2019-12-19 10:43:30 UTC)
cc92081fcf064d1bc835b6f7178d557b (2019-12-19 10:41:12 UTC)
3dfdefdeb1f6d18d4d4eeb0bf9201747 (2019-12-19 10:40:46 UTC)
d493d1faf72e74a09a9b0c03b6f3d0d2 (2019-12-19 10:40:22 UTC)
```

Calling the [v1 endpoint for the search](https://api.bintray.com/conan/czoido/conan-packages/v1/conans/dummy_lib_test_revs/0.1/_/_/search), each time a different `recipe_hash` is returned.
But the `install` command will call to the [download_urls endpoint](https://api.bintray.com/conan/czoido/conan-packages/v1/conans/dummy_lib_test_revs/0.1/_/_/download_urls) and that one is always returning the latest revision.

The [endpoint for getting the manifest](https://api.bintray.com/conan/czoido/conan-packages/v1/conans/dummy_lib_test_revs/0.1/_/_/digest) retrieves the last revision always as well.

In conclusion, it seems to be a bug in the v1 search endpoint for bintray when the packages were uploaded using revisions but consumed without revisions enabled in the client.

**Comment by czoido** [comment]
More tests, using **Artifactory Version 6.12.0** this time. Uploading multiple revisions to a local Artifactory:

```
➜  conan search dummy_lib_test_revs/0.1@ -r=artifactory-local --revisions
Revisions for 'dummy_lib_test_revs/0.1' at remote 'artifactory-local':
b60d4707a2bf89f1978c6d9c1ade65cb (2019-12-20 14:37:28 UTC)
3e7652088be12cd887e4aef2dd569aed (2019-12-20 14:37:24 UTC)
9f04352cede02c9a803d89a8b6fdcde2 (2019-12-20 14:37:21 UTC)
2b0a2a2ea70fd03e7103f4ffd882a1d8 (2019-12-20 14:37:17 UTC)
f11543adae484526d83e615383121c06 (2019-12-20 14:37:15 UTC)
24a2da36e61781901adfa23436fffb08 (2019-12-20 14:37:11 UTC)
08e038453f068647b8505caf3345d18e (2019-12-20 14:37:08 UTC)
6742c4f352ec23b74feb31f60756e379 (2019-12-20 14:37:05 UTC)
81bd024d58f4ec8ae0f5a97f71299054 (2019-12-20 14:37:02 UTC)
ea3770aea04c49d049995183a4a31f8e (2019-12-20 14:36:59 UTC)
d0f2297211fccff599e18a6554e1bbbc (2019-12-20 14:36:55 UTC)
a501396a607ec4d7b36cf86078f4aecb (2019-12-20 14:36:52 UTC)
ddca89fc083be173d190175c2e26c9ea (2019-12-20 14:36:49 UTC)
03c8f4dc1691ebe861ae19f62df1885c (2019-12-20 14:36:46 UTC)
29a026bb82911e8f3bde19395c4a16a0 (2019-12-20 14:36:43 UTC)
eaaff6330b93109dc0b2994d29779d1e (2019-12-20 14:36:40 UTC)
```

Calling the v1 endpoint for search (in this case: http://localhost:8090/artifactory/api/conan/conan-local/v1/conans/dummy_lib_test_revs/0.1/_/_/search) return:

```
{
  "103f6067a947f366ef91fc1b7da351c588d1827f" : {
    "settings" : {
      "os" : "Macos",
      "compiler.libcxx" : "libc++",
      "arch" : "x86_64",
      "compiler" : "apple-clang",
      "build_type" : "Release",
      "compiler.version" : "11.0"
    },
    "options" : {
      "shared" : "False"
    },
    "requires" : [ ],
    "recipe_hash" : "f11543adae484526d83e615383121c06"
  }
}
```
Which is the revision: `f11543adae484526d83e615383121c06 (2019-12-20 14:37:15 UTC)` which is not the last one. Now it retreives always the same revision (it is not random) but not the last one. So it os going to **falsely mark the recipe as outdated from the package**.

For the other endpoints, everything seems correct.

**Comment by czoido** [comment]
Closing the issue. Progress of this bug will be done in: https://www.jfrog.com/jira/browse/RTFACT-20919
